src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -181,6 +185,9 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
     val (resolvedIndexedColumns, resolvedIncludedColumns) = resolveConfig(df, indexConfig)
     val columnsFromIndexConfig = resolvedIndexedColumns ++ resolvedIncludedColumns
 
+    val escapedIndexedColumns = SchemaUtils.prefixNestedFieldNames(resolvedIndexedColumns)
+    val escapedIncludedColumns = SchemaUtils.prefixNestedFieldNames(resolvedIncludedColumns)
+
```

Comment:
It's not straight forward. `select` needs a head and the unpacked tail, while `toDF` unpacks the full collection. Also, a while ago in previous implementation I used the `SchemaUtils.prefixNestedFieldNames(allIndexColumns)` to keep the symmetry but in a review I was suggested to change it to the way it is right now.

I'll try putting it in a nice form.

Reply:
I see. If we decided to make `resolve` return `isNestedField` (https://github.com/microsoft/hyperspace/pull/379#discussion_r599159093), we can do `col(config).as(escaped)` only for nested columns, so that `toDF` is not required?